src/Loading/Plugins/babylonFileLoader.ts

```diff
@@ -589,21 +590,28 @@ SceneLoader.RegisterPlugin({
                         }
 
                         // Material ?
-                        if (parsedMesh.materialId) {
-                            var materialFound = (loadedMaterialsIds.indexOf(parsedMesh.materialId) !== -1);
+                        if (parsedMesh.materialUniqueId || parsedMesh.materialId) {

```

Comment:
There are several locations where `materialUniqueId` is assumed to not be equal to 0, but it is indeed possible that `materialUniqueId=0` (`Material.uniqueId` (==`materialUniqueId`) is assigned with a call to `Scene.getUniqueId()` which is using `UniqueIdGenerator.UniqueId` which starts at 0).

Maybe the easiest fix would be to make `UniqueIdGenerator.UniqueId` starts at 1...

Reply:
Sure. Is this change okay @deltakosh ?

Reply:
yes no problem